### ISSUE TITLE
after installing new version & resetting settings - database does full rescan on startup

### ISSUE THREAD

**Issue by filevans** [droidmjt/Droidsound#234]
Also I've got a really good idea, when resetting settings after installing a new version, could it ask you: also delete database? Yes/no.

This way you choose not to lose your db. As when testing new versions you don't need to lose your db if you made no changes to your music folder...
And just wastes time deleting it then having to rescan
@droidmjtdroidmjt
about 1 hour ago
what database you are referring to? the songs.db in droidsound folder?
@filevansfilevans
27 minutes ago
The database where all your music collection is stored for browsing and searching in droidsound. This doesn't need to be deleted when resetting settings, unless you specifically request it
@droidmjtdroidmjt
21 minutes ago
It should be already so that if you install a new version and your current DB version (database structure version) is up2date, it won't create a new songs.db, unless you check the checkbox from welcome message box (i will add that)
@filevansfilevans
7 minutes ago
Yes but when you install a new version you get problems if you don't reset everything and restart. And when you do you lose your database, it would be better if asked you if you want to delete database too, when it says : are you sure you want to delete all settings, a tick box below (unticked by default) 'also delete database'

I'm not sure which deletes database, when you restore GUI or reset settings
@droidmjtdroidmjt
3 minutes ago
full rescan deletes database, nothing else
@droidmjtdroidmjt
2 minutes ago
new install only deletes your preferences, maybe i can add a switch

**Comment by filevans** [comment]
No I don't mean when installing, I mean after when you go to settings and reset settings or restore GUI, I'm not sure which deletes database but one of then does

**Comment by droidmjt** [comment]
what happens is that, when you click reset preferences, and restart the app, it thinks its fresh install (cannot find DB version) and performs full scan

**Comment by filevans** [comment]
Why can't it find the db. Hasn't it been deleted?

**Comment by filevans** [comment]
If not deleted then what about 'recover database' option, automatically better

**Comment by droidmjt** [comment]
if you reset preferences and then restore gui, then close the app and use filebrowser to see droidsound/songs.db, it will be there, it is not deleted. When ever i install a new version, it just refreshes some things but will never do FULL rescan, it's not possible UNLESS the songs.db is missing.

**Comment by droidmjt** [comment]
do you have lots of local files unarchived or do you use archives to store your files?

**Comment by droidmjt** [comment]
im upload a new version now, it doesn't have the checkbox yet but try this, don't reset preferences or guis or remove anything, just start installing the new version, see what happens.

**Comment by droidmjt** [comment]
https://www.dropbox.com/s/2z2z6jii20ne2vu/droidsounde16647_12.apk?dl=0

**Comment by droidmjt** [comment]
the problem here is that i have hard time in reproducing your scan scenario

**Comment by droidmjt** [comment]
>> what happens is that, when you click reset preferences, and restart the app, it thinks its fresh install (cannot find DB version) and performs full scan

ignore that comment, even in that case it won't perform full rescan, i've read the init code now few times and there is now full rescan triggering anywhere.

**Comment by filevans** [comment]
i see, it doesnt delete the db

but before as the default setting was ticked to "full rescan on startup" so if you delete your settings and restart after installing a new version it forces a full rescan, there was the problem, so now youve default that setting unticked so thats resolved it then, so i can close this case

**Comment by filevans** [comment]
if you wanted to be really technical you could set it to detect if the database is there, and it not there trigger full scan, otherwise not, but personally i dont mind doing manual scan

**Comment by filevans** [comment]
it is still doing full scan on startup, after resetting settings, closing program and restarting....

default is ticked to on, this doesnt need to happen if you have already dont a scan on the previous installed version and  already have a full db file...

**Comment by droidmjt** [comment]
i need to put more local files and see if this is happening

**Comment by filevans** [comment]
definitely happening, every time, just set scan database setting default off - unticked, that is ok for now, and then if someone hasnt installed the program before then they do a manual scan...

**Comment by droidmjt** [comment]
the topic is misleading, what exactly is the problem now? tell me in details how to reproduce the problem and what do you want to get fixed

**Comment by filevans** [comment]
ok, i have full database already scanned right?

then i install a new version to test, right?

then, i load it and in settings i go to "reset settings" because sometimes previous settings cause things not to work properly on a new version

then i close and restart program

THEN IT DOES FULL RESCAN

this is not needed, i already have a full scanned database from previously installed version

please can you change default setting not to scan on startup.... as when you reset settings and restart it defaults to full scan on start up and does a massive redundant scan..

**Comment by filevans** [comment]
all youve got to do it change default setting to off - untick the option to do full scan on start up

this way when installing a new version it will use the previous database, no need for a full scan on start up...

**Comment by droidmjt** [comment]
how about this, you install version x01, you use it, you disable rescan on start from settings, then you install version x02, it checks if you have disabled the rescan flag from settings, if yes, it won't do any rescanning ?

**Comment by filevans** [comment]
no. already explained that, because after resetting settings which you have to do it defaults back to ticked, i keep telling you that

you have to reset settings for new versions, otherwise you get problems

**Comment by filevans** [comment]
if youre not happy to do that, cant you just set the option to check if the database file exists, and if not do full rescan on start up, and if does exist dont scan. thats the other option

**Comment by droidmjt** [comment]
ah so preferences must be resetted, i see...i'll think of something

**Comment by filevans** [comment]
ive got another idea, on first time start up it asks you:

do you want to scan your your device and add to database?
yes / no

that resolves it

**Comment by droidmjt** [comment]
fixed, i changed the way how it detects whether its first run or not, next install will do another scan but after that, it won't, promise :)

**Comment by droidmjt** [comment]
also, you need to disable the rescan on start checkbox, this is because some users might want to use the rescan on start

**Comment by droidmjt** [comment]
i need to favour those who wish to use "rescan on start" + i need to know version infos, so how it goes from here, you need to do one more rescan, after that, just disable the rescan from settings, next time when you install a new version, it won't do rescan because it sees your DB + it knows the versions are correct + you unchecked the rescan box

**Comment by droidmjt** [comment]
https://www.dropbox.com/s/txwcryjefda73e7/droidsounde16647_13.apk?dl=0

**Comment by droidmjt** [comment]
crap i missed one version check, i'll upload new version soon

**Comment by droidmjt** [comment]
https://www.dropbox.com/s/yre7ghmkvgfb8dh/droidsounde16647_14.apk?dl=0

**Comment by filevans** [comment]
Yes tested that, reinstalled and reset settings twice and only scanned the first time. Good

Being as the new setting is based on detection, you can remove that option in settings now to do full scan on startup... As it's now a redundant option

**Comment by droidmjt** [comment]
there is no such thing as full scan on startup, i will keep it there and you still need to untick it, but only once since this info will be stored forever in your DB, close this case

**Comment by filevans** [comment]
So tell me why the option needs to be there to scan database on startup?
Now it detects if database file is there and does scan depending on that

**Comment by filevans** [comment]
And surely if you untick the option, then press reset settings, then restart, then the setting would default to ticked again. Then would scan on startup... That's logical. Which we don't want.

Too much hasle and confusion keep having to untick that, after installing new versions, should just detect if database file exists and scan or not based on that
 No need other options

**Comment by droidmjt** [comment]
>>And surely if you untick the option, then press reset settings, then restart, then the setting would default to ticked again. Then would scan on startup... That's logical. Which we don't want.

Not true, when you untick the box, the value gets saved to the database, even if you reset preferences thousand times it won't remove the value from the DB, so next time when a new version gets installed, this value will be checked, the value will be kept in the DB as long as you either delete the DB or perform manual full rescan

**Comment by droidmjt** [comment]
DS-E has 2 rescan modes, FULL and non-FULL, the non-FULL scan checks orphaned entries from DB and removes them, and is way quicker than the FULL scan.

**Comment by droidmjt** [comment]
in your case, the non-full scan takes long time because you probably have lots of local files, if you keep them in an archive, this wouldn't happen

**Comment by filevans** [comment]
Yes I have complete SID, SC68, SNDH and very big mod collection

**Comment by filevans** [comment]
Yes I have complete SID, SC68, SNDH and very big mod collection

**Comment by filevans** [comment]
too much to store sid collection in archives, so many folders and files, but i could store sndh, any sc68, still would cause problems if i need to send one file by email to you or someone else it makes it more hassle if everything is in one archive

**Comment by droidmjt** [comment]
i think this case is done, the rescan problem is no more there

**Comment by filevans** [comment]
just going to test and install new version now for 100% certainty

**Comment by filevans** [comment]
yes good, thanks very much, good job

**Comment by filevans** [comment]
this is still happening, and its annoying, ive just uninstalled completely the previous version and reinstalled and upon first run it scans everything, why cant this stop happening, i just want to start the program without a scan, my idea about coming with a dialogue box is the best, upon first run a box comes up and says:

----------------------------------------------------------------------

droidsound would like to scan your device now for music
[ OK ] ------- [ cancel scan ]

----------------------------------------------------------------------

**Comment by droidmjt** [comment]
this is my bad, i changed something badly

**Comment by droidmjt** [comment]
https://www.dropbox.com/s/2r2ocss1f4w71n1/droidsounde16647_24.apk?dl=0

**Comment by filevans** [comment]
Thanks will test everything

**Comment by filevans** [comment]
ok i think this is resolved, but i dont want to uninstall or reset all my settings now to fully test it, ill reopen if i find any problem, thank you
